presto-main/src/main/java/io/prestosql/execution/AddColumnTask.java

```diff
@@ -71,7 +71,7 @@ public String getName()
             WarningCollector warningCollector)
     {
         Session session = stateMachine.getSession();
-        QualifiedObjectName tableName = createQualifiedObjectName(session, statement, statement.getName());
+        QualifiedObjectName tableName = metadata.redirectTable(session, createQualifiedObjectName(session, statement, statement.getName()), warningCollector);
```

Comment:
Agreed. But making separate calls makes `tableName` not "effectively final", so it can't be referenced in lambdas. In this case, we have to make changes around lambda calls that seem irrelevant, e.g, wiping out lambda calls or declaring a new final reference. So it's like a trade-off between readabilities of code and commit. What do you think?